crypto.go

```diff
@@ -0,0 +1,59 @@
+package skynet
+
+import (
+	"encoding/binary"
+	"errors"
+	"golang.org/x/crypto/blake2b"
+	"golang.org/x/crypto/ed25519"
+)
+
+// encodeNumber converts the given number into a byte array.
+func encodeNumber(number uint64) []byte {
+	b := make([]byte, 8)
+	binary.LittleEndian.PutUint64(b, number)
+	return b
+}
+
+// encodeString converts the given string into a byte array.
+func encodeString(toEncode string) []byte {
+	encodedNumber := encodeNumber(uint64(len(toEncode)))
+	return append(encodedNumber, []byte(toEncode)...)
+}
+
+// hashDataKey hashes the given data key.
+func hashDataKey(dataKey string) []byte {
+	encodedDataKey := encodeString(dataKey)
+	hash := blake2b.Sum256(encodedDataKey)
+	return hash[:]
+}
+
+// hashAll takes all given arguments and hashes them.
+func hashAll(args ...[]byte) []byte {
+	var bytes []byte
+	for _, arg := range args {
+		bytes = append(bytes, arg...)
+	}
+	hash := blake2b.Sum256(bytes)
+	return hash[:]
+}
+
+// hashRegistryEntry hashes the given registry entry.
+func hashRegistryEntry(s RegistryEntry) []byte {
```

Comment:
Can you please add a test for this function? We have one in `skynet-js` you can copy.

Reply:
I would use the encoding package `encoding.Marshl`  That would eliminate the need for the majority of this file I believe.